### PR TITLE
upgrade etcd to v3.4.3

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve? 
Fix problem: 
- raft prevote bug https://github.com/etcd-io/etcd/issues/10355

### What is changed and how it works?
upgrade depend.

**Note:** Cannot downgrade to lower version.
### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,4 @@ default.pd
 tags
 /.retools/
 default*
+!defaults.go
```
Comment (disksing) [inline .gitignore]: can we change `default*` to `default.pd*`?

Comment (nolouch) [reply .gitignore]: Good idea.

File: server/api/utiletcdhttpapi.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ import (
 
 const (
 	//etcd peer detail API
-	//return struct https://github.com/coreos/etcd/blob/master/etcdserver/stats/server.go
+	//return struct https://go.etcd.io/etcd/blob/master/etcdserver/stats/server.go
```
Comment (rleungx) [inline server/api/utiletcdhttpapi.go]: The URL is invalid.

File: server/server.go [hunk 3]
```diff
@@ -130,7 +130,7 @@ func CreateServer(cfg *Config, apiRegister func(*Server) http.Handler) (*Server,
 		// The etcd master version has removed embed.Config.SetupLogging.
 		// Now logger is set up automatically based on embed.Config.Logger,
 		// Use zap logger in the test, otherwise will panic.
-		// Reference: https://github.com/coreos/etcd/blob/master/embed/config_logging.go#L45
+		// Reference: https://go.etcd.io/etcd/blob/master/embed/config_logging.go#L45
```
Comment (rleungx) [inline server/server.go]: ditto

File: server/leader.go [hunk 4]
```diff
@@ -338,7 +338,7 @@ func (s *Server) watchLeader(leader *pdpb.Member, revision int64) {
 			}
 
 			for _, ev := range wresp.Events {
-				if ev.Type == mvccpb.DELETE {
+				if int32(ev.Type) == int32(mvccpb.DELETE) {
```
Comment (rleungx) [inline server/leader.go]: Why do we need to use `int32`?

**Pull request merged**